mtags-shared/src/main/scala/scala/meta/internal/metals/utils/LimitedFilesManager.scala

```diff
@@ -0,0 +1,42 @@
+package scala.meta.internal.metals.utils
+
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+
+class LimitedFilesManager(
+    directory: Path,
+    fileLimit: Int,
+    prefixPattern: String
+) {
+  private val fileNameRegex = s"${prefixPattern}([-+]?[0-9]+)".r
+
+  def getAllFiles(): List[TimestampedFile] = {
+    if (Files.exists(directory) && Files.isDirectory(directory)) {
+      directory.toFile.listFiles().flatMap(timestampedFile).toList
```

Comment:
Just a heads up, this will load entries for all files from a directory into the memory at once. Alternative would be to walk the filesystem and collect the 10 newest log files, then stream the folder contents one-by-one and delete any log file that doesn't match one of those 10. But maybe this simpler solution is enough. :slightly_smiling_face:

Reply:
For `logs` we can only have one above the limit at a time. For reports, in theory many can be created but we avoid creating the same report more than once... So I don't think that would ever be problematic.